OpenRA.Mods.TS/Traits/Buildings/LowBridge.cs

```diff
@@ -24,27 +25,39 @@ class LowBridgeInfo : ITraitInfo, Requires<BuildingInfo>
 		public object Create(ActorInitializer init) { return new LowBridge(init.Self, this); }
 	}
 
-	class LowBridge : INotifyCreated
+	class LowBridge : INotifyAddedToWorld, INotifyRemovedFromWorld
 	{
 		readonly LowBridgeInfo info;
+		readonly IEnumerable<CPos> cells;
 
 		public LowBridge(Actor self, LowBridgeInfo info)
 		{
 			this.info = info;
+
+			var buildingInfo = self.Info.TraitInfo<BuildingInfo>();
+			cells = FootprintUtils.PathableTiles(self.Info.Name, buildingInfo, self.Location);
 		}
 
-		void INotifyCreated.Created(Actor self)
+		void UpdateTerrain(Actor self, byte terrainIndex)
 		{
-			var buildingInfo = self.Info.TraitInfo<BuildingInfo>();
-			var cells = FootprintUtils.PathableTiles(self.Info.Name, buildingInfo, self.Location);
-			var tileSet = self.World.Map.Rules.TileSet;
-			var terrainIndex = tileSet.GetTerrainIndex(info.TerrainType);
 			foreach (var cell in cells)
 				self.World.Map.CustomTerrain[cell] = terrainIndex;
 
 			var domainIndex = self.World.WorldActor.TraitOrDefault<DomainIndex>();
 			if (domainIndex != null)
 				domainIndex.UpdateCells(self.World, cells);
 		}
+
+		void INotifyAddedToWorld.AddedToWorld(Actor self)
+		{
+			var tileSet = self.World.Map.Rules.TileSet;
+			var terrainIndex = tileSet.GetTerrainIndex(info.TerrainType);
+			UpdateTerrain(self, terrainIndex);
+		}
+
+		void INotifyRemovedFromWorld.RemovedFromWorld(Actor self)
```

Comment:
You'll need to trigger a check (probably in a FrameEndTask) to kill any actors that are no longer allowed in the cell.  Otherwise you end up with orphaned units where they don't belong:

![screen shot 2016-08-26 at 16 59 58](https://cloud.githubusercontent.com/assets/167819/18011871/d9f2a310-6bae-11e6-8c77-39e3b0674aaa.png)